lms/static/js/student_account/views/RegisterView.js

```diff
@@ -127,6 +127,37 @@
                         jsHook: this.authWarningJsHook,
                         message: fullMsg
                     });
+                },
+
+                getFormData: function() {
+                    var obj = FormView.prototype.getFormData.apply(this, arguments),
+                        $form = this.$form,
+                        $label,
+                        $emailElement,
+                        $confirmEmailElement,
+                        email = '',
+                        confirmEmail = '';
+
+                    $emailElement = $form.find('input[name=email]');
+                    $confirmEmailElement = $form.find('input[name=confirm_email]');
+
+                    if ($confirmEmailElement) {
+                        email = $emailElement.val();
+                        confirmEmail = $confirmEmailElement.val();
+                        $label = $form.find('label[for=' + $confirmEmailElement.attr('id') + ']');
+
+                        if (email !== confirmEmail) {
+                            this.errors.push('<li>' + $confirmEmailElement.data('errormsg-required') + '</li>');
```

Comment:
Blech. So, it looks like the logic for required fields is handled around the same time as this, so I get two "confirm your email" messages if I leave the field blank. If you make sure to not do the check if `confirmEmail.val() === ""`, then the "required" logic will handle the message if it's blank, and we won't get duplicates.